airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -618,7 +615,8 @@ def await_pod_completion(
             remote_pod = self.read_pod(pod)
             if remote_pod.status.phase in PodPhase.terminal_states:
                 break
-            if istio_enabled and container_is_completed(remote_pod, container_name):
+            if container_is_completed(remote_pod, container_name):
+                self.log.info("Base container %s has completed", container_name)
```

Comment:
Working around this issue, I added a 30 second timeout after this condition is met to see if the sidecar container exits normally. Without the 30 second timeout, I was seeing cases where the sidecar was being terminated before gracefully exiting and that seemed to cause some errors.

Reply:
What is the container that's being run in that case? [The default behavior from k8s is to send a `SIGTERM`](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination) and other `STOP_SIGNAL`s configured on containers should be respected but I wouldn't be surprised if you hit an edge case.

Reply:
FWIW we workaround the issue by just enabling the istio proxy along with running an HTTP server in another container. The current behavior with an istio proxy matches the behavior here except without the need for the istio proxy. As expected, when the base container succeeds our HTTP server container gets a SIGTERM and exits as expected.

Reply:
> What is the container that's being run in that case?

Tried to document here https://github.com/apache/airflow/issues/39625 - it was the GCS FUSE sidecar in Google Cloud Composer. I believe there to be a bug with that sidecar as well. This PR seems like a great step to mitigate the issue on the airflow side.